src/commands/deploy.ts

```diff
@@ -243,19 +244,20 @@ export default class Deploy extends DeployCommand {
       dependency_manager = await LocalDependencyManager.create(this.app.api);
       dependency_manager.environment = env_config;
 
+      this.setExtraEnvironmentVariables(flags.parameter, env_config);
+
       const extra_interfaces = this.getExtraInterfaces(flags.interface);
-      this.updateEnvironmentInterfaces(env_config, extra_interfaces, parsed_component_version.namespaced_component_name);
+      const edges = (await dependency_manager.getGraph()).edges;
+      this.updateEnvironmentInterfaces(dependency_manager.environment, extra_interfaces, edges);
```

Comment:
I would have expected the interfaces to be set in the env config before the dep-manager/graph is created. Editing the edges directly doesn't feel right.

Reply:
How else would we get dependencies and their interfaces to map back to without generating the graph?